### DISCUSSION TITLE
Get order of changed fields

### DISCUSSION THREAD

**Discussion by pedroR** [react-hook-form/react-hook-form#8641] [General]
Good morning. 
I am using the `watch()` to get the values of several fields, so I specify the fields name inside `watch`:  
```
const one = watch('field1'); 
const two = watch('field2'); 
const nine = watch('field9'); 
```

I could use `const multiple=watch(["field1", "field2"...."field9"])` but for now I prefer to get them isolated. 

Then I call a function to update an array
```
const result = [];
const updateList = (fieldValue) => {
      result.push(fieldValue);
 };

updateList(one);
updateList(two);
updateList(nine);

```

However I would like to keep the order of changed values  by the user. 
If the user change by this order one -> nine -> two. I would like to get: `result = [valueOne, valueNine, valueTwo]` instead of `result = [valueOne, valueTwo, valueThree]`

Could anyone help how to accomplish the order of changed fields?

**Comment by bluebill1049** [comment]
I would do that on `onChange` on each input and track the order there.
